src/Symfony/Component/Validator/Tests/Constraints/WordCountValidatorTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\WordCount;
+use Symfony\Component\Validator\Constraints\WordCountValidator;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+use Symfony\Component\Validator\Tests\Constraints\Fixtures\StringableValue;
+
+class WordCountValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator(): WordCountValidator
+    {
+        return new WordCountValidator();
+    }
+
+    /**
+     * @dataProvider provideValidValues
+     */
+    public function testValidWordCount(string|\Stringable|null $value, int $expectedWordCount)
+    {
+        $this->validator->validate($value, new WordCount(min: $expectedWordCount, max: $expectedWordCount));
+
+        $this->assertNoViolation();
+    }
+
+    public function testTooShort()
+    {
+        $constraint = new WordCount(min: 4, minMessage: 'myMessage');
+        $this->validator->validate('my ascii string', $constraint);
+
+        $this->buildViolation('myMessage')
+            ->setParameter('{{ count }}', 3)
+            ->setParameter('{{ min }}', 4)
+            ->assertRaised();
+    }
+
+    public function testTooLong()
+    {
+        $constraint = new WordCount(max: 3, maxMessage: 'myMessage');
+        $this->validator->validate('my beautiful ascii string', $constraint);
+
+        $this->buildViolation('myMessage')
+            ->setParameter('{{ count }}', 4)
+            ->setParameter('{{ max }}', 3)
+            ->assertRaised();
+    }
+
+    /**
+     * @dataProvider provideInvalidTypes
+     */
+    public function testNonStringValues(mixed $value)
+    {
+        $this->expectException(UnexpectedValueException::class);
+        $this->expectExceptionMessageMatches('/Expected argument of type "string", ".*" given/');
+
+        $this->validator->validate($value, new WordCount(min: 1));
+    }
+
+    public static function provideValidValues()
```

Comment:
```suggestion
    public static function provideValidValues(): iterable
```

Reply:
Should we? Nicolas seemed to disagree on this point, hence this choice: https://github.com/symfony/symfony/pull/53154#discussion_r1436499093

Reply:
Oh ok, I was only aware of all the `void` noise. I personally add iterable to my providers and also describe it via PHPDoc 🤷‍♂️

Reply:
Yeah, it's a Nicolas thing. 🙂